playbooks/secrets-init.yml

```diff
@@ -52,13 +52,9 @@
             gs_mode: "plugin"
           when: vault_init | default('') == 'plugin'
 
-        - name: Populate secrets.yml with missing secrets
-          ansible.builtin.shell: "{{ base_folder }}/scripts/generate-secret.sh -s {{ item }} -m {{ gs_mode }} >> {{ secrets_file }}"
-          changed_when: true
-          when: not lookup('vars', item, default=False)
-          tags:
-            - molecule-idempotence-notest
-          loop:
+        - name: Generate loop of mandatory passwords when empty inventory is provided
```

Comment:
what about dropping this task and `when` on the next one and adding `|| (groups['all'] | length == 0)` to all the new following tasks?

Reply:
you can also set a fact `empty_inventory` to avoid repeating it and also make the intention more clear

Reply:
Yeah it'll be better